templates/java/org/yarp/Loader.java.erb

```diff
@@ -30,15 +30,26 @@ public class Loader {
         byte[] get(ByteBuffer buffer, int oneBasedIndex) {
             int index = oneBasedIndex - 1;
             byte[] constant = cache[index];
+
             if (constant == null) {
                 int offset = bufferOffset + index * 8;
                 int start = buffer.getInt(offset);
                 int length = buffer.getInt(offset + 4);
 
                 constant = new byte[length];
-                System.arraycopy(source, start, constant, 0, length);
+
+                if ((start & (1 << 31)) == 0) {
```

Comment:
Another way to check this in Java is `(start < 0)` since `int` is signed 32-bit

Reply:
But since this works and mirrors the Ruby code, feel free to keep it as-is (same below)

Reply:
Updated to use `Integer.compareUnsigned`